store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedOperationHandler.java

```diff
@@ -0,0 +1,112 @@
+ */
+@Since("2.0.0")
+public class FederatedOperationHandler<INPUT, OUTPUT> implements OperationHandler<FederatedOperation<INPUT, OUTPUT>> {
+
+    public static final String ERROR_WHILE_RUNNING_OPERATION_ON_GRAPHS = "Error while running operation on graphs";
+
+    @Override
+    public Object doOperation(final FederatedOperation<INPUT, OUTPUT> operation, final Context context, final Store store) throws OperationException {
+        final Iterable<?> allGraphResults = getAllGraphResults(operation, context, (FederatedStore) store);
+
+        return mergeResults(allGraphResults, operation, (FederatedStore) store);
+    }
+
+    private Iterable getAllGraphResults(final FederatedOperation<INPUT, OUTPUT> operation, final Context context, final FederatedStore store) throws OperationException {
+        try {
+            List<Object> results;
+            final Collection<Graph> graphs = getGraphs(operation, context, store);
+            results = new ArrayList<>(graphs.size());
+            for (final Graph graph : graphs) {
+
+                final Operation updatedOp = FederatedStoreUtil.updateOperationForGraph(operation.getUnClonedPayload(), graph);
+                if (updatedOp != null) {
+                    try {
+                        if (updatedOp instanceof Output) {
+                            results.add(graph.execute((Output) updatedOp, context));
+                        } else {
+                            graph.execute(updatedOp, context);
+                            if (nonNull(operation.getMergeFunction())) {
+                                results.add(null);
+                            }
+                        }
+                    } catch (final Exception e) {
+                        if (!operation.isSkipFailedFederatedExecution()) {
+                            throw new OperationException(FederatedStoreUtil.createOperationErrorMsg(operation, graph.getGraphId(), e), e);
+                        }
+                    }
+                }
+            }
+
+            return results;
+        } catch (final Exception e) {
+            throw new OperationException(ERROR_WHILE_RUNNING_OPERATION_ON_GRAPHS, e);
+        }
+
+    }
+
+    private Object mergeResults(final Iterable resultsFromAllGraphs, final FederatedOperation operation, final FederatedStore store) throws OperationException {
+        try {
+            Object rtn = null;
+
+            final BiFunction mergeFunction = nonNull(operation.getMergeFunction()) ? operation.getMergeFunction() : store.getDefaultMergeFunction();
+
+            //Reduce
+            for (final Object resultFromAGraph : resultsFromAllGraphs) {
+                rtn = mergeFunction.apply(resultFromAGraph, rtn);
+            }
+
+            return rtn;
+        } catch (final Exception e) {
+            throw new OperationException(String.format("Error while merging results. %s", Objects.toString(e.getMessage(), ""), e));
```

Comment:
There is a typo here. As it is now, `e` is not part of the call to `OperationException` and the stack trace is entirely missing as a result.
```suggestion
            throw new OperationException(String.format("Error while merging results. %s", Objects.toString(e.getMessage(), "")), e);
```